src/app/tabs/newAdmin/operations.js

```diff
@@ -1,8 +1,82 @@
-import { toggleBasicAdvanced } from './actions';
+import Web3 from 'web3';
+import { keccak_256 as sha3 } from 'js-sha3';
 
-export const start = () => (dispatch) => {
+import {
+  rskOwner as rskOwnerAddress,
+  registrar as tokenRegistrarAddress,
+} from '../../adapters/configAdapter';
+import { rskOwnerAbi, tokenRegistrarAbi } from './abis.json';
+import { gasPrice as defaultGasPrice } from '../../adapters/gasPriceAdapter';
+
+import {
+  toggleBasicAdvanced, checkIfSubdomain, requestCheckTokenOwner, receiveCheckTokenOwner,
+  errorCheckTokenOwner, requestFifsMigrationStatus, receiveFifsMigrationStatus,
+  errorFifsMigrationStatus,
+} from './actions';
+
+const web3 = new Web3(window.ethereum);
+
+export const checkIfSubdomainOrTokenOwner = domain => async (dispatch) => {
+  const labelsAmount = domain.split('.').length;
+
+  if (labelsAmount > 2) {
+    dispatch(checkIfSubdomain(true));
+  }
+
+  const label = domain.split('.')[0];
+  const accounts = await window.ethereum.enable();
+  const currentAddress = accounts[0];
+
+  dispatch(requestCheckTokenOwner());
+
+  const hash = `0x${sha3(label)}`;
+
+  const rskOwner = new web3.eth.Contract(
+    rskOwnerAbi, rskOwnerAddress, { gasPrice: defaultGasPrice },
+  );
+
+  rskOwner.methods.ownerOf(hash).call((error, result) => {
+    if (error) {
+      dispatch(errorCheckTokenOwner());
+      return;
+    }
+
+    const domainOwner = result;
+
+    dispatch(receiveCheckTokenOwner(
+      domainOwner.toLowerCase() === currentAddress.toLowerCase(),
+      domainOwner,
+    ));
+  });
+};
+
+export const checkIfFIFSRegistrar = domain => async (dispatch) => {
+  dispatch(requestFifsMigrationStatus());
+
+  return new Promise((resolve) => {
+    const label = `0x${sha3(domain.split('.')[0])}`;
+
+    const tokenRegistrar = new web3.eth.Contract(
+      tokenRegistrarAbi, tokenRegistrarAddress, { gasPrice: defaultGasPrice },
+    );
+
+    tokenRegistrar.methods.entries(label).call((error, result) => {
+      if (error) {
+        dispatch(errorFifsMigrationStatus());
+      }
+
+      const deed = result[1];
+
+      return resolve(dispatch(receiveFifsMigrationStatus(deed === '0x0000000000000000000000000000000000000000')));
+    });
+  });
+};
+
+export const start = domain => (dispatch) => {
   const showAdvancedView = localStorage.getItem('adminAdvancedView');
   dispatch(toggleBasicAdvanced(showAdvancedView === 'true'));
+  dispatch(checkIfSubdomainOrTokenOwner(domain));
+  dispatch(checkIfFIFSRegistrar(domain));
 };
 
 export const toggleBasicAdvancedSwitch = showAdvancedView => (dispatch) => {
```

Comment:
Is this copy-paste? Where does it come from? Shouldn't we reuse the copied function?
> When removing old UI, we should `git mv` file and not change dir manually. This makes better experience for reviews.

Reply:
Sorta, these functions (above on line 19 and 53) somewhat resemble the functionality used in the /admin folder but not 100%. They are cleaned up for consistent variable naming.

We could link to the operations file in the /admin folder for some of these, but then it would save in the admin reducer and some use the old notification system which I'm not using anymore. When we go to remove the old admin it will get very messy as there will be multiple operations files.

My thought was to start clean and have the code all it one place to begin with rather than cleaning up later.

Reply:
Great! Let's leave them as this.